core/src/main/scala/org/apache/spark/scheduler/TaskSetExcludeList.scala

```diff
@@ -163,3 +167,18 @@ private[scheduler] class TaskSetExcludelist(
     }
   }
 }
+
+private[scheduler] object TaskSetExcludelist {
+
+  /**
+   * Returns true if the excludeOnFailure is enabled on the task/stage level,
+   * based on checking the configuration in the following order:
+   * 1. Is taskset level exclusion specifically enabled or disabled?
+   * 2. Is overall exclusion feature enabled or disabled?
+   * 3. Default is off
+   */
+  def isExcludeOnFailureEnabled(conf: SparkConf): Boolean = {
```

Comment:
Shouldn't we check the overall exclusion feature first?

Reply:
My thought is to have the taskset/application level configs as the new standard knobs, and treat `spark.excludeOnFailure.enabled` as a legacy fallback for backward compatibilities.